### DISCUSSION TITLE
EBSCO integration

### DISCUSSION THREAD

**Discussion by ie76** [vufind-org/vufind#2708] [Q&A] [answered]
Hello, 

i'd like to integrate EBSCO api, i have set my credentials in EDS.ini file. But unfortunately it says that i encounter some issues in the frontend. 

My question is if there's any other solution i may use? 
And if the EDS will be integrated in the search globally.

Thanks.

**Comment by demiankatz** [comment, marked as answer]
@ie76, if you fill in the EDS.ini settings, you should be able to perform searches at the /EDS endpoint under your VuFind base URL (e.g. here is a link to Villanova's version: https://library.villanova.edu/Find/EDS). If you only want to use VuFind for EDS searching, you can change the defaultModule setting in config.ini to make EDS your default search. If you want to combine different types of searches, see this wiki page: https://vufind.org/wiki/configuration:combining_search_types

If none of that helps and you're still seeing an error, you might want to put VuFind into development mode, or try one of our other standard troubleshooting techniques, to narrow down what is going wrong. See: https://vufind.org/wiki/development:troubleshooting

If after troubleshooting you have a more specific error message but aren't sure how to resolve it, please feel free to post more details here. Let me know if there's anything that requires further clarification, or if I can do anything else to help!

**Comment by ie76** [comment]
Thank you for your usual fast reply. In fact, i have solved the issue by setting ip_auth to false. Now i'd like to combine the searches in order to have them all in one search using the api. However, i'm still struggling now to try and make work swagger bypassing "Permission denied". 

Thank you for your answer 🙏.

**Comment by demiankatz** [comment]
@ie76, have you looked at permissions.ini? This is the file which controls the rules that grant various permissions within the system. You probably need to enable the API in order to get Swagger working. There is an example configuration here which will allow API access to localhost: https://github.com/vufind-org/vufind/blob/dev/config/vufind/permissions.ini#L165-L171 -- you can adjust the settings to make access broader if necessary. Let me know exactly what you need if you're not easily able to figure it out on your own and I can help with more details.

**Comment by ie76** [comment]
Hello @demiankatz well i'm trying to request the api from swagger from http://localhost:4567/vufind/swagger-ui/#/Search/get_search unfortunately i have as response 
`{
  "status": "ERROR",
  "statusMessage": "Permission denied"
}`
i have uncommented the settings like so: 
`
; Search and Record API permissions.
[api.SearchAndRecord]
permission[] = access.api.Search
permission[] = access.api.Record
require = ANY
ipRange[] = '127.0.0.1'
ipRange[] = '::1'
`
but still nothing.

**Reply by demiankatz** [reply]
Is there any possibility that you edited $VUFIND_HOME/config/vufind/permissions.ini instead of $VUFIND_LOCAL_DIR/config/vufind/permissions.ini? The configs under $VUFIND_HOME are the defaults that get overridden when matching files are copied into $VUFIND_LOCAL_DIR. It's an easy mistake to edit the wrong version. If that doesn't help you, let me know and I'll look into this further.

**Reply by ie76** [reply]
Hello @demiankatz i have edited indeed the file $VUFIND_HOME/config/vufind/permissions.ini, i noticed no changes. 
Then i have copied the file into $VUFIND_HOME/local/config/vufind and still not. 
I have copied as well the file into $VUFIND_LOCAL_DIR/config/vufind/ but still not. 

I'm still struggling to find out how to fix this issue. 
I'm using Vagrant.

**Reply by demiankatz** [reply]
I wonder if Vagrant's virtual networking is representing your IP differently than expected by the default configuration.

What happens if you add:

```
role[] = guest
role[] = loggedin
```

To your permissions.ini section? This should enable ALL users to access the API, regardless of source. Since you're only doing this in a test environment, there should not be a security disadvantage to an open configuration. If that gets things working and you want to keep a more restrictive configuration, we can do further troubleshooting to figure out what is wrong... but if it enables you to test what you need to test, it may be good enough for the moment.

**Reply by ie76** [reply]
When i added 
``` 
role[] = guest
role[] = loggedin
```
it worked thank you very much.

**Comment by ie76** [comment]
Another question, it maybe out of topic, i can open one if it's the case, but is the 9.0 available now ? Because the Blended Search is very interesting in my case.

**Reply by demiankatz** [reply]
The Blended Search functionality is available if you check out the dev branch, but the 9.0 release is not quite ready yet -- I expect it will likely be released in April, though an official date has not been announced yet.

**Reply by ie76** [reply]
Thank you for your reply. 🙏 
Is the EDS search can be returned in swagger search api ?

**Reply by demiankatz** [reply]
No, the Swagger search API is only for accessing VuFind's internal Solr index. If you want to interact programmatically with EDS, it would be more efficient to use the EDS API directly.

**Reply by ie76** [reply]
Understood, in the next version 9.0, in blended search. We will be able to fetch data from Solr and EDS maybe ? or it will be justo for the indexes existing in Solr ?

**Reply by EreMaijala** [reply]
Blended search can search in EDS as well, and VuFind comes with basic configuration for it.

**Reply by ie76** [reply]
It will be available in the api as well ?

**Reply by demiankatz** [reply]
No, the API will only return Solr data. Blended search is only available through the user interface at this time.

**Reply by ie76** [reply]
Thank you for the reply 🙏
